### PR TITLE
add utility for browsing and removing messages for dlq

### PR DESCRIPTION
**Pull request by Iamfrenzy**

### CODE REVIEW

File: test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DlqMessageClient.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    private final Sleeper sleeper = new Sleeper();
+
+
+    public DlqMessageClient()  {
+
+        initialise();
+    }
+
+    @VisibleForTesting
+    public DlqMessageClient(QueueConnection connection, Session session, Queue dlqQueue, String dlqName,
```
Comment (allanmckenzie) [inline test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DlqMessageClient.java]: parameters should be final

File: test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/QueueUriProvider.java [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+import static uk.gov.justice.services.test.utils.common.host.TestHostProvider.getArtemisHost;
 
 public class QueueUriProvider {
 
     private static final String BASE_URI_PATTERN = "tcp://%s:61616";
 
     public String getQueueUri() {
-        return format(BASE_URI_PATTERN, getHost());
+        return format(BASE_URI_PATTERN, getArtemisHost());
     }
```
Comment (allanmckenzie) [inline test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/QueueUriProvider.java]: I'm afraid you can't change this - there are a lot of integration tests that rely on getting the correct url to speak to the deployed app. You'll need to create another class or method

File: test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DlqMessageClient.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        } catch (JMSException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+
+    public List<JsonObject> browse() {
+        try {
+            List<JsonObject> messages = new ArrayList<>();
+            sleeper.sleepFor(200);
```
Comment (allanmckenzie) [inline test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DlqMessageClient.java]: Not sure sleep is a good idea here.

File: test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DlqMessageClient.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            dlqBrowser = session.createBrowser(dlqQueue);
+            Enumeration enumeration = dlqBrowser.getEnumeration();
```
Comment (allanmckenzie) [inline test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DlqMessageClient.java]: final

File: test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DlqMessageClient.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+
+    public void removeMessages() {
+        try {
+            dlqMessageConsumer = session.createConsumer(dlqQueue);
+            while (receiveNoWait().isPresent()) {
+            }
+        } catch (JMSException e) {
+            LOGGER.error("Fatal error cleaning messges from DLQ");
+            throw new RuntimeException(e);
```
Comment (allanmckenzie) [inline test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DlqMessageClient.java]: put the error message into the exception

**Review by purple52 — CHANGES_REQUESTED** [review]
This appears to replicate a lot of the code in the main MessageConsumerClient, and does its own connection creation, etc. I would prefer this to reuse the existing client code (refactored/extended where appropriate). We also need examples for use cases in the class Javadocs and Javadocs for all public methods. Please come and talk to us and perhaps pair with one of us to produce a plan for how to integrate these classes.

File: test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DlqMessageClient.java (hunk 1 above)
Comment (Iamfrenzy) [reply test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DlqMessageClient.java]: done

File: test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/QueueUriProvider.java (hunk 2 above)
Comment (Iamfrenzy) [reply test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/QueueUriProvider.java]: done

File: test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DlqMessageClient.java (hunk 3 above)
Comment (Iamfrenzy) [reply test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DlqMessageClient.java]: done

File: test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DlqMessageClient.java (hunk 4 above)
Comment (Iamfrenzy) [reply test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DlqMessageClient.java]: done

File: test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DeadLetterQueueBrowser.java [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    /**
+     * clean up resources
+     */
+    public void close() {
+        jmsSessionFactory.close();
+    }
+
+    private void doClose(final AutoCloseable closeable) {
```
Comment (purple52) [inline test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/DeadLetterQueueBrowser.java]: Is this still used? If not, it should be removed

**Review by purple52 — CHANGES_REQUESTED** [review]
I've added one comment to the code. Also please check code coverage; Coveralls is reporting it has dropped by 1.6%.
Before merge, we'll need to squash your commits into one. If you know how to do that please go for it. Otherwise, I can do it when I merge.

**Review by purple52 — APPROVED** [review]

**Pull request merged**